pkg/bytecode/symbol.go

```diff
@@ -18,6 +21,17 @@ type Symbol struct {
 // SymbolTable is a mapping of string identifiers to symbols.
 type SymbolTable struct {
 	store map[string]Symbol
+	// index keeps track of the number of locals inside this scope,
+	// an enclosed SymbolTable will inherit a starting index from
+	// its outer table.
+	index int
+	// nestedMaxIndex is the maximum index of any inner symbol
+	// tables. It propagates up to the top-level symbol table
+	// for a function or block-scoped globals.
```

Comment:
I think we use the word "global" differently. In the VM I understand it to mean: top level variables available everywhere excluding top-level block-scoped variables. So this comment confuses me a little as `globals` seem to include block-scoped top-level variables.

Might be worth while to step back and write a clean description on the purpose and workings of the `SymbolTable` as it seems to be such a crucial piece in understanding how all variables and scopes and their handling inside the compiler and VM hang together.

(summarizing an IRL discussion with @camh- who's volunteered to provide such a description if i haven't misunderstood).

Reply:
Done. I've taken your description of SymbolTable and added that and re-worded this comment to remove "block-scope globals"